src/penalized_losses.jl

```diff
@@ -0,0 +1,63 @@
+# Note (1). See
+# https://discourse.julialang.org/t/weight-regularisation-which-iterates-params-m-in-flux-mutating-arrays-is-not-supported/64314
+
+
+""" Penalizer(λ, α)
+
+Returns a callable object `penalizer` for evaluating regularization
+penalties associated with some numerical array. Specifically,
+`penalizer(A)` returns
+
+   λ*(α*L1 + (1 - α)*L2),
+
+where `L1` is the sum of absolute values of the elments of `A` and
+`L2` is the sum of squares of those elements.
+"""
+struct Penalizer{T}
+    lambda::T
+    alpha::T
+    function Penalizer(lambda, alpha)
+        lambda == 0 && return new{Nothing}(nothing, nothing)
+        T = promote_type(typeof.((lambda, alpha))...)
+        return new{T}(lambda, alpha)
+    end
+end
+
+(::Penalizer{Nothing})(::Any) = 0
+function (p::Penalizer)(A)
+    λ = p.lambda
+    α = p.alpha
+    # avoiding broadcasting; see Note (1) above
+    L2 = sum(x^2 for x in A)
+    L1 = sum(abs(x) for x in A)
```

Comment:
You could try `sum(abs2, A)` and `sum(abs, A)` as well. Zygote and ChainRules have special-cased rules for `sum(f, x)`, so those might be faster (I haven't benchmarked).

Reply:
Thanks! That seems to work.